app.py

```diff
@@ -312,14 +315,15 @@ def log_event(event_type):
         return event_type
 
 if __name__ == "__main__":
-    app.secret_key = "super sedcret"
-    compile_sass()
     if app.debug:
+        print("Debug mode is on; autogenerating new secret key.")
+        app.secret_key = ''.join(random.choice(string.ascii_uppercase + string.ascii_lowercase + string.digits) for _ in range(20))
         xray_recorder.configure(sampling=False)
         from livereload import Server, shell
         server = Server(app.wsgi_app)
         server.watch('./static/assets/scss/*', compile_sass)
         server.watch('./')
         server.serve(host='0.0.0.0',open_url=False,port=5000,debug=True)
     else:
+        app.secret_key = environ["APP_SECRET_KEY"]
```

Comment:
The point of the secret key is to protect our sessions (stored through cookies) [from being modified by the end user](https://en.wikipedia.org/wiki/Session_poisoning).

This code here is a bit dangerous because if `APP_SECRET_KEY` isn't defined on environ, it'll throw a key error, but also because if the setting is empty, people will have an easy time cracking the sessions.

You're right about live reload; every time a new secret key is generated, existing sessions become invalid.

Reply:
I'll just add the APP_SECRET_KEY to our normal secret store and it'll be pulled in on deploy time, no extra config needed - just like the DB_ENDPOINT. Shouldn't affect any local dev. Added a more cogent exception case.
I think it should cause an exception if a secret key isn't provided, no?